core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -90,14 +90,15 @@ public boolean hasNext() {
         if (mEntry != null) {
           return true;
         }
-        while (it.hasNext()) {
+        if (it.hasNext()) {
           mEntry = it.next();
-          // Do not journal the root mount point.
-          if (!mEntry.getKey().equals(ROOT)) {
-            return true;
-          } else {
+          // Skip the root mount point, which is considered a part of initial state, not journaled
```

Comment:
Is it weird that this method knows that the root mount point is special (part of initial state), but the `clear` method doesn't know the root mount point is special?